packages/ra-core/src/controller/saveContext/useMutationMiddlewares.ts

```diff
@@ -51,12 +51,11 @@ export const useMutationMiddlewares = <
         []
     );
 
-    const mutateWithMiddlewares = useCallback(
-        (
-            fn: MutateFunc,
-            ...args: Parameters<MutateFunc>
-        ): ReturnType<MutateFunc> => {
-            let index = callbacks.current.length - 1;
+    const getMutateWithMiddlewares = useCallback((fn: MutateFunc) => {
+        // Stores the current callbacks in a closure to avoid losing them if the calling component is unmounted
+        const currentCallbacks = [...callbacks.current];
```

Comment:
Code looks good. However I'm wondering if we are still at risk of a regression regarding optimistic cases, since this part of the code (storing the callbacks in a closure) is not covered by automated tests.
Do you think we could cover that part too?

Reply:
Nice! :muscle: